components/org.wso2.carbon.consent.mgt.ui/src/main/resources/web/consent/add-purpose.jsp

```diff
@@ -79,6 +85,22 @@
         if (localClaims != null) {
             claims.addAll(Arrays.asList(localClaims));
         }
+
+        String currentUser = (String) session.getAttribute("logged-user");
+        ConsentManagementServiceClient serviceClient = new ConsentManagementServiceClient(currentUser);
+
+        JSONArray purposeIdListParsed = new JSONArray(purposeIdList);
+        for (int i = 0; i < purposeIdListParsed.length(); i++) {
+            Purpose retrievedPurpose = serviceClient.getPurpose(Integer.parseInt(purposeIdListParsed.get(i).toString()));
```

Comment:
serviceClient.getPurpose(purposeIdListParsed.get(i));

Reply:
No need to toString and convert to int again

Reply:
when we parsed a string in format '[1,2,3,4]' to a JSONArray using JSONparser the array contains list of objects. not strings. In that purposeIdListParsed.get(i) cannot be used. element should be converted to an Integer

Reply:
changed to serviceClient.getPurpose(purposeIdListParsed.getInt(i)) [ since serviceClient.getPurpose(purposeIdListParsed.get(i)) returns an object but need Integer type]